WordPress/src/main/java/org/wordpress/android/ui/media/services/MediaUploadService.java

```diff
@@ -274,34 +231,38 @@ private void cancelUpload() {
     }
 
     private void dispatchUploadAction(@NonNull final MediaModel media) {
-        AppLog.i(T.MEDIA, "Dispatching upload action for media with local id: " + media.getId() +
+        AppLog.i(AppLog.T.MEDIA, "Dispatching upload action for media with local id: " + media.getId() +
                 " and path: " + media.getFilePath());
         media.setUploadState(UploadState.UPLOADING.name());
         mDispatcher.dispatch(MediaActionBuilder.newUpdateMediaAction(media));
 
         MediaPayload payload = new MediaPayload(mSite, media);
         mDispatcher.dispatch(MediaActionBuilder.newUploadMediaAction(payload));
-
-        if (mListener != null) {
-            mListener.onUploadBegin(mCurrentUpload);
-        }
     }
 
     private void dispatchCancelAction(@NonNull final MediaModel media) {
-        AppLog.i(T.MEDIA, "Dispatching cancel upload action for media with local id: " + media.getId() +
+        AppLog.i(AppLog.T.MEDIA, "Dispatching cancel upload action for media with local id: " + media.getId() +
                 " and path: " + media.getFilePath());
         MediaPayload payload = new MediaPayload(mSite, mCurrentUpload);
         mDispatcher.dispatch(MediaActionBuilder.newCancelMediaUploadAction(payload));
     }
 
+    private void completed(){
+        AppLog.i(AppLog.T.MEDIA, "Media Upload Service > completed");
+        if (mQueue == null || mQueue.size() == 0) {
```

Comment:
Okay, we can tidy up the duplication later if it's still relevant to do so 👍.

Could we rename `completed()` to something more descriptive in the meantime though? Something like `stopServiceIfUploadsComplete()`?